lib/mix/phoenix/schema.ex

```diff
@@ -142,13 +144,21 @@ defmodule Mix.Phoenix.Schema do
     |> to_string()
   end
 
-  @doc """
-  Fetches the unique attributes from attrs.
-  """
-  def uniques(attrs) do
-    attrs
-    |> Enum.filter(&String.ends_with?(&1, ":unique"))
-    |> Enum.map(& &1 |> String.split(":", parts: 2) |> hd |> String.to_atom)
+  def extract_attr_flags(cli_attrs) do
+    {attrs, uniques, redacts} = Enum.reduce(cli_attrs, {[], [], []}, fn attr, {attrs, uniques, redacts} ->
+      prefix1 = byte_size(attr) - 7
+      prefix2 = byte_size(attr) - 14
+      attr_name = attr |> String.split(":") |> hd() |> String.to_atom()
+      case attr do
+        <<attr::size(prefix2)-binary, ":redact:unique">> -> {[attr | attrs], [attr_name | uniques], [attr_name | redacts]}
+        <<attr::size(prefix2)-binary, ":unique:redact">> -> {[attr | attrs], [attr_name | uniques], [attr_name | redacts]}
+        <<attr::size(prefix1)-binary, ":unique">> -> {[attr | attrs], [attr_name | uniques], redacts}
+        <<attr::size(prefix1)-binary, ":redact">> -> {[attr | attrs], uniques, [attr_name | redacts]}
+        _ -> {[attr | attrs], uniques, redacts}
+      end
```

Comment:
What do you think about a function that was recursive and did this:

```
defp split_flags(["unique" | rest], name, attrs, uniques, redacts),
  do: split_flags(rest, name, attrs, [name | uniques], redacts)

defp split_flags(["redact" | rest], name, attrs, uniques, redacts),
  do: split_flags(rest, name, attrs, uniques, [name | redacts])

defp split_flags(rest, _name, attrs, uniques, redacts),
  do: {Enum.join(Enum.reverse(rest), ":"), uniques, redacts}
```

and we would call it like this:

```
[attr_name | rest] = String.split(":")
attr_name = String.to_atom(attr_name)
split_flags(Enum.reverse(rest), attr_name, attrs, uniques, redacts)
```

?